server/src/main/java/org/elasticsearch/search/aggregations/Aggregator.java

```diff
@@ -152,9 +152,43 @@ public BucketComparator bucketComparator(String key, SortOrder order) {
     }
 
     /**
-     * Build an aggregation for data that has been collected into {@code bucket}.
+     * Build an aggregation for data that has been collected into
+     * {@code owningBucketOrd}.
+     * @deprecated use {@link #buildAggregations(long[])} instead 
      */
-    public abstract InternalAggregation buildAggregation(long bucket) throws IOException;
+    @Deprecated
+    public InternalAggregation buildAggregation(long owningBucketOrd) throws IOException {
+        /*
+         * Temporarily check if it looks like we're being called from a test
+         * and try to answer with the top level agg. This just prevents us from
+         * having to modify 1231234134124 tests in one PR.
+         */
```

Comment:
I think we should do this on the same pr, otherwise we should add a no-release tag so that we don't forget to do it before a release.

Reply:
I agree here. I don't think the change to calling buildTopLevel() in all the tests will be too distracting

Reply:
<img src="https://media.giphy.com/media/OCu7zWojqFA1W/giphy.gif">

Reply:
It was only about 40 places! I thought there were a couple hundred. Easy!